File: Programming Basics/JavaScript/Nested loops/Lab/sum-of-two-numbers.js
function solve(input) {
    const start = Number(input[0]);
    const end = Number(input[1]);
    const magicNumber = Number(input[2]);

    let isFound = false;
    let order = 0;
    for (let i = start; i <= end; i++) {
        for (let j = start; j <= end; j++) {
            order++;

            if (i + j == magicNumber) {
                console.log(`Combination N:${order} (${i} + ${j} = ${magicNumber})`);
                isFound = true;
                break;
            }
        }

        if (isFound) break;
    }

    if (!isFound) console.log(`${order} combinations - neither equals ${magicNumber}`);
}

File: Programming Basics/C++/Nested loops/Labs/SumOfTwoNumbers/SumOfTwoNumbers.cpp
#include <iostream>
using namespace std;

int main()
{
    int start, end, magicNumber;
    cin >> start >> end >> magicNumber;

    bool isFound = false;
    int order = 0;
    for (int i = start; i <= end; i++)
    {
        for (int j = start; j <= end; j++)
        {
            order++;

            if (i + j == magicNumber)
            {
                cout << "Combination N:" << order << " (" << i << " + " << j << " = " << magicNumber << ")" << endl;
                isFound = true;
                break;
            }
        }

        if (isFound) break;
    }

    if (!isFound)
        cout << order << " combinations - neither equals " << magicNumber << endl;
}
